website/docs/Guides-Csharp/authoring-languages.md

```diff
@@ -0,0 +1,241 @@
+---
+title: Authoring Languages
+sidebar_position: 11
+---
+
+# Authoring a LionWeb Language in C#
+
+This guide demonstrates how to define and serialize a LionWeb language dynamically using the `LionWeb-CSharp` library.
+
+## Overview
+
+In this example, we define a simple language called **Complaints Language** with the following components:
+
+- A **language** entity
+- A **Complaints Book** concept that contains complaints
+- A **Complaint** concept with properties:
+  - Date (optional)
+  - Description (required)
+  - Urgent flag (required boolean)
+- A **Date** primitive type
+- The **serialization** of the language into a JSON file using LionWeb v2024_1
+
+## Required Namespaces
+
+```csharp
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using LionWeb.Core;
+using LionWeb.Core.M1;
+using LionWeb.Core.M3;
+using LionWeb.Core.VersionSpecific.V2024_1;
+```
+
+## Defining the Language
+
+```csharp
+var language = new DynamicLanguage("lang", LionWebVersions.v2024_1)
+{
+    Key = "key-lang", 
+    Version = "0",
+    Name = "Complaints Language",
```

Comment:
This is not a valid name -- we defined all `IKeyed` names must be valid programming identifiers

Reply:
Right